File: BackEnd/src/main/java/com/example/demo/entity/Ownership.java
package com.example.demo.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "ownership")
public class Ownership extends JpaEntity implements Serializable {

    private String country;
    private String city;
    private String address;
    private String house_type;
    private Integer rooms;
    private Integer bathrooms;
    private String description;
    private Double price;
    private String payment_conditions;
    private String[] additional_services;
    private String[] additional_fees;
    private String[] images;
    private String policies_cancellation;
    private Double latitude;
    private Double longitude;
    private Boolean pets_allowed;
    private String smoking_policy;

    @Column(nullable = false)
    private UserEntity userId;

}

File: FrontEnd/src/app/home/interfaces/Ownership.ts
import { Service } from "./Service";

export enum TypeOfHouse {
    APARTMENT, 
    HOUSE,
    CABIN
}

export interface Ownership {
    id: string,
    createAt: null,
    modifiedAt: null,
    country: string,
    state: string,
    city: string,
    address: string,
    house_type: TypeOfHouse,
    images: string[],
    rooms: number,
    bathrooms: number,
    house_area: number,
    description: string,
    price: number,
    deposit: number,
    payment_conditions: string,
    additional_services: Service[],
    additional_fees: [],
    policies_cancellation: string,
    latitude: number,
    longitude: number,
    pets_allowed: boolean,
    smoking_policy: string,
    available_date: number
}